alphagov/paas-cf
# [#153119909] Upgrade Ruby to 2.5

## What

Update Ruby version to 2.5

We decided to use the same (latest) Ruby version everywhere. We still used 2.2 in multiple places which reached EOL on 2018-03-31 (#1), so we decided to migrate to the latest stable version.

We had to upgrade govuk-lint as older versions didn't support Ruby 2.5. Thanks to the upgrade we've got new style check violations but we decided to ignore them (at least for now), as this would unnecessarily blow up the story and increase the risk.

By upgrading the cf-cli Docker images there is a side-effect as the CF cli was upgraded to 6.34.1 (https://github.com/alphagov/paas-docker-cloudfoundry-tools/pull/112) but we never updated the image tags in the pipelines. Probably it's better if we clean this up, so I applied some fixes (see commits) because of the CLI upgrade too.

#1 https://www.ruby-lang.org/en/downloads/branches/

## How to review

1. Create a new CF env from this branch

    ```
    BRANCH=upgrade_ruby_153119909 SELF_UPDATE_PIPELINE=false ENABLE_DATADOG=true make dev pipelines
    ```

1. Make sure all tests pass, including the ones in the health group.

1. Run the create-cloudfoundry pipeline again on this env or on your existing env.

1. Destroy the CF environment. (do the bosh cli tests from the paas-bootstrap PR first)

## How to merge

❗️ The PR contains a WIP commit which needs to be updated with the final Docker image tags.

## Who can review

Not me.
